src/coreclr/nativeaot/Runtime/MiscHelpers.cpp

```diff
@@ -51,11 +51,11 @@ EXTERN_C NATIVEAOT_API void __cdecl RhSpinWait(int32_t iterations)
     ASSERT(iterations > 0);
 
     // limit the spin count in coop mode.
-    ASSERT_MSG(iterations <= 10000 || !ThreadStore::GetCurrentThread()->IsCurrentThreadInCooperativeMode(),
+    ASSERT_MSG(iterations <= 1024 || !ThreadStore::GetCurrentThread()->IsCurrentThreadInCooperativeMode(),
```

Comment:
This change is just because we switched to SpinWait that is calibrated to 35 nsec. 
Previous numbers were assumed pre-skylake, or simply put the numbers that the underlying API would divide by 8.   
I also made it power of 2. It is more convenient.